verifiers/internal/gha/verifier.go

```diff
@@ -43,68 +44,79 @@ func verifyEnvAndCert(env *dsse.Envelope,
 	provenanceOpts *options.ProvenanceOpts,
 	builderOpts *options.BuilderOpts,
 	defaultBuilders map[string]bool,
-) ([]byte, string, error) {
+) ([]byte, *utils.BuilderID, error) {
 	/* Verify properties of the signing identity. */
 	// Get the workflow info given the certificate information.
 	workflowInfo, err := GetWorkflowInfoFromCertificate(cert)
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
 	// Verify the workflow identity.
 	builderID, err := VerifyWorkflowIdentity(workflowInfo, builderOpts,
 		provenanceOpts.ExpectedSourceURI, defaultBuilders)
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
 	// Verify properties of the SLSA provenance.
 	// Unpack and verify info in the provenance, including the Subject Digest.
 	provenanceOpts.ExpectedBuilderID = builderID
 	if err := VerifyProvenance(env, provenanceOpts); err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
 	fmt.Fprintf(os.Stderr, "Verified build using builder https://github.com%s at commit %s\n",
 		workflowInfo.JobWobWorkflowRef,
 		workflowInfo.CallerHash)
 	// Return verified provenance.
 	r, err := base64.StdEncoding.DecodeString(env.Payload)
-	return r, builderID, err
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Temporary code.
+	bid, _ := utils.BuilderIDNew(builderID)
+	return r, bid, nil
 }
 
 // VerifyArtifact verifies provenance for an artifact.
 func (v *GHAVerifier) VerifyArtifact(ctx context.Context,
 	provenance []byte, artifactHash string,
 	provenanceOpts *options.ProvenanceOpts,
 	builderOpts *options.BuilderOpts,
-) ([]byte, string, error) {
+) ([]byte, *utils.BuilderID, error) {
 	rClient, err := rekor.NewClient(defaultRekorAddr)
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
 	/* Verify signature on the intoto attestation. */
 	env, cert, err := VerifyProvenanceSignature(ctx, rClient, provenance, artifactHash)
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
-	return verifyEnvAndCert(env, cert,
+	content, builderID, err := verifyEnvAndCert(env, cert,
 		provenanceOpts, builderOpts,
 		defaultArtifactTrustedReusableWorkflows)
+	if err != nil {
```

Comment:
nit: why did this need to change from returning those results? (I guess we have trust in github not providing malformed job_workflow_refs so you may not even be able to get a verified but malformed builder, but the assertion maybe should be there)

Reply:
not actually needed. A residual change I should have removed. Reverted. Good catch.